### DISCUSSION TITLE
v2.0.1

### DISCUSSION THREAD

**Discussion by shuding** [vercel/swr#2366] [New]
## What's Changed
* types: make FullConfiguration generic by @huozhi in https://github.com/vercel/swr/pull/2273
* type: enforce array literal as tuple by @promer94 in https://github.com/vercel/swr/pull/2283
* Improve type test by @promer94 in https://github.com/vercel/swr/pull/2289
* update bunchee to remove the extra target cli option by @huozhi in https://github.com/vercel/swr/pull/2298
* fix: useSWRMutation should always use the latest fetcher by @koba04 in https://github.com/vercel/swr/pull/2290
* types: conditional swr response by @huozhi in https://github.com/vercel/swr/pull/2301
* fix: types properites in exports fields by @koba04 in https://github.com/vercel/swr/pull/2341
* Update license year by @IcedMonk in https://github.com/vercel/swr/pull/2344
* feat: add all mutate options support to useSWRInfinite by @koba04 in https://github.com/vercel/swr/pull/2350
* ci: add publint to check packaging on CI by @koba04 in https://github.com/vercel/swr/pull/2363

## New Contributors
* @IcedMonk made their first contribution in https://github.com/vercel/swr/pull/2344

**Full Changelog**: https://github.com/vercel/swr/compare/2.0.0...v2.0.1

<hr /><em>This discussion was created from the release <a href='https://github.com/vercel/swr/releases/tag/v2.0.1'>v2.0.1</a>.</em>

**Comment by nstepien** [comment]
I tried `2.0.1` but I don't get the type improvements in our code, for example:
```ts
interface Data {}

export function useData() {
  const obj = useSWR<Data>('data', { suspense: true });

  console.log(obj.data, obj.isLoading);
}
```

Here `obj.data` is of type `Data | undefined` instead of `Data`, and `obj.isLoading` is `boolean`.
If I remove `<Data>` then `obj.isLoading` is correctly `false` but then `obj.data` is `any`.
